gateway/src/integration-test/java/com/hurence/webapiservice/http/grafana/ImportJsonEndPointIT.java

```diff
@@ -0,0 +1,109 @@
+        HttpWithHistorianSolrITHelper
+                .initWebClientAndHistorianSolrCollectionAndHttpVerticleAndHistorianVerticle(client, container, vertx, context);
+        webClient = HttpITHelper.buildWebClient(vertx);
+        assertHelper = new AssertResponseGivenRequestHelper(webClient, "/api/grafana/query");
+    }
+
+    @AfterAll
+    public static void afterAll(Vertx vertx, VertxTestContext context) {
+        webClient.close();
+        vertx.close(context.succeeding(rsp -> context.completeNow()));
+    }
+
+    @Test
+    @Timeout(value = 5, timeUnit = TimeUnit.SECONDS)
+    public void testNewAdd(Vertx vertx, VertxTestContext testContext) {
+
+        assertAddRequestGiveResponseFromFile(vertx, testContext,
+                "/http/grafana/query/extract-algo/testAdd/request.json",
+                "/http/grafana/query/extract-algo/testAdd/expectedResponse.json");
+        assertRequestGiveResponseFromFile(vertx, testContext,
+                "/http/grafana/query/extract-algo/test0/request.json",
+                "/http/grafana/query/extract-algo/test0/expectedResponse.json");
+    }
+
+    @Test
+    @Timeout(value = 5, timeUnit = TimeUnit.SECONDS)
+    public void testWrongAdd(Vertx vertx, VertxTestContext testContext) {
+
+        assertWrongAddRequestGiveResponseFromFile(vertx, testContext,
+                "/http/grafana/query/extract-algo/testWrongAdd/request.json");
+    }
+
+    public void assertRequestGiveResponseFromFile(Vertx vertx, VertxTestContext testContext,
+                                                  String requestFile, String responseFile) {
+        assertHelper.assertRequestGiveArrayResponseFromFile(vertx, testContext, requestFile, responseFile);
+    }
+
+    public void assertAddRequestGiveResponseFromFile(Vertx vertx, VertxTestContext testContext,
+                                                  String requestFile, String responseFile) {
+        final FileSystem fs = vertx.fileSystem();
+        Buffer requestBuffer = fs.readFileBlocking(AssertResponseGivenRequestHelper.class.getResource(requestFile).getFile());
+        webClient.post("/historian-server/ingestion/json")
+                .as(BodyCodec.jsonObject())
+                .sendBuffer(requestBuffer.getDelegate(), testContext.succeeding(rsp -> {
+                    testContext.verify(() -> {
+                        assertEquals(200, rsp.statusCode());
+                        assertEquals("OK", rsp.statusMessage());
+                        JsonObject body = rsp.body();
+                        Buffer fileContent = fs.readFileBlocking(AssertResponseGivenRequestHelper.class.getResource(responseFile).getFile());
+                        JsonObject expectedBody = new JsonObject(fileContent.getDelegate());
+                        assertEquals(expectedBody, body);
+                        testContext.completeNow();
+                    });
+                }));
+    }
```

Comment:
I think this method exist already somewhere else. You should not duplicate general method. Otherwise if we want to change it we have to do it for all the methods. It's easy to just copy paste, but in the long run it is not good.

Reply:
ok done